_drafts/wsl2-as-a-firmware-development-environment.md

```diff
@@ -0,0 +1,485 @@
+          Kernel: Linux 6.8.0-36-generic
+    Architecture: x86-64
+```
+
+All other platforms are compared to the Ubuntu results because Ubuntu is the
+fastest.
+
+| Metric                  | Result     | Compared to Native Ubuntu |
+| ----------------------- | ---------- | ------------------------- |
+| Duration                | 11m 4.444s | -                         |
+| Average build duration  | 2.723s     | -                         |
+| Workload multiplier     | 1.000      | -                         |
+| Duration of a 40s build | 40.000s    | -                         |
+
+![Screenshot of the native Ubuntu benchmark run]({% img_url zephyr-development-environments/ubuntu-twister-run.png %})
+
+### WSL2 Ubuntu 24.04
+
+```
+  Virtualization: wsl
+Operating System: Ubuntu 24.04 LTS
+          Kernel: Linux 5.15.153.1-microsoft-standard-WSL2
+    Architecture: x86-64
+```
+
+| Metric                  | Result                    | Compared to Native Ubuntu |
+| ----------------------- | ------------------------- | ------------------------- |
+| Duration                | 12m 34.642s               | + 1m 30.642s              |
+| Average build duration  | 3.092s                    | + 0.369s                  |
+| Workload multiplier     | 754.642s / 664.444s       | * 1.136                   |
+| Duration of a 40s build | 40.000s * 1.136 = 45.440s | + 5.440s                  |
+
+![Screenshot of the WSL2 5.15 benchmark run]({% img_url zephyr-development-environments/wsl2-twister-run-2.png %})
+
+I also compiled Kernel 6.1 and 6.6 for WSL2 and saw that Kernel 6.1 shrank the
+performance gap by about 4%.  In my opinion, that's not worth it, but it shows
+that WSL2's [upcoming upgrade](https://github.com/microsoft/WSL2-Linux-Kernel/releases)
+to Kernel 6.6 may come with some more performance gains.
+
+```
+  Virtualization: wsl
+Operating System: Ubuntu 24.04 LTS
+          Kernel: Linux 6.1.21.2-microsoft-standard-WSL2+
+    Architecture: x86-64
+```
+
+![Screenshot of the WSL2 6.1 benchmark run]({% img_url zephyr-development-environments/wsl2-twister-run-kernel6.1.png %})
+
+### Windows 11
+
+> Tip: One of the main things that slows down build systems in Windows is Windows Defender.  Build
+systems do a ton of file IO and Windows Defender scans go wild during the build.  For
+this test, I disabled Windows Defender entirely, which you should never do.  Instead,
+you should painstakingly ignore your build folders and toolchains in
+Defender settings and pray for better support from Microsoft someday!
+
+> Note: Windows 11 wasn't able to properly complete the benchmark due
+to issues with long paths generated by the build system.  I've compiled Zephyr
+projects on Windows many times without (too many) problems, but this issue
+does give me pause. Well, that, and it being half the speed of WSL2, at best!
+
+> Note: I made a few patches to improve Twister’s Windows support which is why
+a diff is shown in the prompt.
```

Comment:
Is the screen shot maybe cut off at the top? I don't see the diff you're referring to

Reply:
Ya know what, I'm gonna remove the comment cause I don't think anyone will notice.  It's the little ~5 in this screenshot.

<img width="113" alt="image" src="https://github.com/user-attachments/assets/8e2f3926-fd2e-4c74-97c2-5292a8c4493d">

If someone questions it then they can get in the comments - engagement !